_posts/2024-06-10-infinispan-15-queries.adoc

```diff
@@ -28,16 +28,7 @@ Following an example of entity, query and result:
 [source,java]
 ----
 @Indexed(index = "play")
-public class Game {
-
-   private final String name;
-   private final String description;
-
-   @ProtoFactory
-   public Game(String name, String description) {
-      this.name = name;
-      this.description = description;
-   }
+public record Game(String name, String description) {
 
    @Keyword(projectable = true, sortable = true)
```

Comment:
You want to get rid of the getters too, and put the annotations on the record field declaration

Reply:
Need to test it first. Then of course I'll do it

Reply:
Example:
```java
@Proto
@Indexed
public record Book(
    @Text String title,
    @Keyword(projectable = true, sortable = true, normalizer = "lowercase", indexNullAs = "unnamed", norms = false) String description,
    int publicationYear,
    Set<Author> authors,
    Type bookType,
    BigDecimal price
) { }
```

Reply:
Tested. It works nicely! Pushed the changes